### DISCUSSION TITLE
rules syntax question

### DISCUSSION THREAD

**Discussion by nkiesel** [Splitties/refreshVersions#531] [Q&A] [answered]
Is there a description about how the rule patterns work?  I think I understand the `^^^` part, but what is e.g. the difference between a patterns `foo:*` and `foo:(-*)`?

I also have a concrete question related to this: I have multiple dependencies on `netty` components like `io.netty:netty-buffer` and `io.netty:netty-resolver:` which should have the same version number.  But I also have  `io.netty:netty` which has a different version number (don't ask me why...).  is it possible to describe this with rules or do I have to keep all the dependencies separate?

**Comment by imashnake0** [comment, marked as answer]
@nkiesel We should definitely have some documentation for it. For now, I would say go through [`androidx-version-alias-rules`](https://github.com/jmfayard/refreshVersions/blob/main/plugins/dependencies/src/main/resources/refreshVersions-rules/androidx-version-alias-rules.txt) and compare it to the [AndroidX ](https://github.com/jmfayard/refreshVersions/blob/main/plugins/dependencies/src/main/kotlin/dependencies/AndroidX.kt)class to see how dependencies can be grouped. Let me know if it's still a little unclear.

**Reply by nkiesel** [reply]
took me a while to start working on this again. I looked at the files you mentioned, but I still am not clear on the `foo:(-*)` syntax. Anything specific i should look at?  Or can you point me at the code which parses this pattern?

**Reply by LouisCAD** [reply]
The parentheses mean it's optional, the dash is required if the part is present. BTW, the parentheses can be omitted in your example since there's always a name (aka. artifact id)

**Reply by nkiesel** [reply]
thx
